### PR TITLE
Small fixes in freighter and wallet

### PR DESCRIPTION
**Pull request by guggero**
A couple of commits extracted out of https://github.com/lightninglabs/taproot-assets/pull/477 that don't necessarily have to do with burning itself.

Fixes https://github.com/lightninglabs/taproot-assets/issues/414.

### CODE REVIEW

File: tapfreighter/wallet.go [hunk 1]
```diff
@@ -515,6 +515,12 @@ func (f *AssetWallet) FundPacket(ctx context.Context,
 
 	totalInputAmt := uint64(0)
 	for _, anchorAsset := range selectedCommitments {
+		// We only use the sum of all assets of the same ID to avoid
+		// counting passive assets as well.
+		if anchorAsset.Asset.ID() != fundDesc.ID {
+			continue
+		}
+
```
Comment (ffranr) [inline tapfreighter/wallet.go:531]: I don't understand this change. Couldn't passive asset also have the same asset ID at this point?

**Review by ffranr — APPROVED** [review]
:+1: Looks good!

Just one question.

File: tapfreighter/wallet.go (hunk 1 above)
Comment (guggero) [reply tapfreighter/wallet.go:531]: Not necessarily. If you look at `removeActiveCommitments`, we declare all assets as "passive" that don't have the same `TapCommitmentKey()`.
But I guess we should use the same comparison here as well, not just the ID. Will update.

**Review by jharveyb — APPROVED** [review]
LGTM! Nice catch re: input value.

**Pull request merged**